File: src/main/java/net/azisaba/townia/data/PlotType.java
package net.azisaba.townia.data;

public enum PlotType {
    DEFAULT,
    SHOP,
    ARENA,
    EMBASSY,
    FARM,
    INN;

    public static PlotType fromString(String s) {
        try {
            return PlotType.valueOf(s.toUpperCase());
        } catch (IllegalArgumentException e) {
            return DEFAULT;
        }
    }
}

File: src/main/java/net/azisaba/townia/data/PlotType.kt
package net.azisaba.townia.data

import java.util.*

enum class PlotType {
    DEFAULT,
    SHOP,
    ARENA,
    EMBASSY,
    FARM,
    INN;

    companion object {
        fun fromString(s: String): PlotType {
            try {
                return valueOf(s.uppercase(Locale.getDefault()))
            } catch (e: IllegalArgumentException) {
                return PlotType.DEFAULT
            }
        }
    }
}